### PR TITLE
[ROOT-8844] Customizable precision via TGeoManager::SetExportPrecisio…

### PR DESCRIPTION
**Pull request by agheata**
…n for geometry ASCII exports

This fixes part of ROOT-8844

### CODE REVIEW

File: geom/gdml/inc/TGDMLWrite.h [hunk 1]
```diff
@@ -219,6 +202,7 @@ class TGDMLWrite : public TObject {
    TString GetPattAxis(Int_t divAxis, const char * pattName, TString& unit);
    Bool_t IsNullParam(Double_t parValue, TString parName, TString objName);
    void UnsetTemporaryBits(TGeoManager * geoMng);
+   void SetFltPrecision(const char *prec) { fFltPrecision = prec; }
```
Comment (pcanal) [inline geom/gdml/inc/TGDMLWrite.h]: Consider std::string_view as the parameter type.

Comment (pcanal) [reply geom/gdml/inc/TGDMLWrite.h]: Actually, shouldn't this just be a UInt_t to both be sturdier and more consistent with the rest of the merge request.

**Comment by Axel-Naumann** [comment]
So writing (and reading) hex floats is not an option? This is an exact, platform independent state dump of an fp value - the best you can get...

**Comment by agheata** [comment]
One of the features of these exports is that they are supposed to be humanly readable, in contrast with binary formats. Don't ask me why, but there are people looking in the GDML after discovering some geometry error...

**Comment by agheata** [comment]
Changed TGDMLWrite interface to take UInt_t. The export precision is now handled by GDML and partially .C export (just matrices for now), but not by xml export. @pcanal, @linev what is your take? Could we do the same for the xml exporter?

**Comment by linev** [comment]
The function name is TBufferText::ComactFloatString(). Some docu [here:](https://root.cern/doc/master/classTBufferText.html#a4017bbd6d7e9a161eeb4af30c7bfabf6) or in the source code:

https://github.com/root-project/root/blob/master/io/io/src/TBufferText.cxx#L696-L789

It is static method, therefore can be used directly.

**Comment by agheata** [comment]
Thanks @linev, I know that there is an issue with very large precision for the file size. I use %.[prec]g which does the expected cutting zeros if not putting too big precision there. It is the user's responsibility to control the precision and implicitly the file size.

**Comment by agheata** [comment]
Merging since what is failing is just un-related tests

**Pull request merged**